vNext/src/Orleans/Shims/AppDomain.cs

```diff
@@ -41,16 +39,9 @@ public Assembly[] GetAssemblies()
                 typeof(Exception).GetTypeInfo().Assembly,
                 typeof(AssemblyProcessor).GetTypeInfo().Assembly,
                 // load up classes for testing, until Assembly scanning is supported in vNext
-                typeof(ClassWithCustomCopier).GetTypeInfo().Assembly,
-                typeof(ClassWithCustomSerializer).GetTypeInfo().Assembly,
-                typeof(SerializerTestClass1).GetTypeInfo().Assembly,
-                typeof(SerializerTestClass2).GetTypeInfo().Assembly,
-                typeof(SerializerTestClass3).GetTypeInfo().Assembly,
-                typeof(SerializerTestClass4).GetTypeInfo().Assembly,
-                typeof(SerializerTestClass5).GetTypeInfo().Assembly,
-                typeof(AsyncObserverArg).GetTypeInfo().Assembly,
-                typeof(StreamSubscriptionHandleArg).GetTypeInfo().Assembly,
-                typeof(GenericArg).GetTypeInfo().Assembly
+                Assembly.Load(new AssemblyName("Orleans.NonSiloTests")),
+                Assembly.Load(new AssemblyName("TestInternalGrains")),
+                Assembly.Load(new AssemblyName("TestGrainInterfaces")) 
```

Comment:
Sure. Loading the nonsilotests assembly might cause issues when we have other test projects, but we can deal with that later. Perhaps by then we have the assembly loader stuff done already. 
Looks good, I'll do a final review tomorrow and merge